integration/docker/main_test.go

```diff
@@ -62,6 +62,20 @@ func TestIntegration(t *testing.T) {
 		}
 	}
 
+	// we need to check that processes like hypervisor, shim and proxy are not running
+	hypervisorPath := KataConfig.Hypervisor[DefaultHypervisor].Path
```

Comment:
I have a feeling, and probably correctly, that @jodh-intel is going to ask if we can somehow factor our the list of things we check to somewhere more 'central'. I'm wondering if it is possible to do this via a yaml or something. It might not be, but the concern will be we are probably going to duplicate this list in a number of places (this is already at least the second place), and if/when we change something later (like add the net monitor or ksm throttler like we did recently, or change the proxy or shim architecture like we did a while back), then we're going to have to locate and modify multiple places (on the upside.... hopefully any change will trigger them to fail, and thus maybe they will be easy to find :-) ).

Maybe we need a little out of the box thinking and brainstorming here. I wonder if we should have a single external command that knows how to sanity check, and we call that command from everywhere (including here, and in the metrics tests). That command could be in `bash`, or `go` - I'm not too fussed.

wdyt @jodh-intel @GabyCT @chavafg 

Other than that - the notion is right - I just want to make sure we centralise and future proof as much as we can (and I'd like to also be checking mounts and /var dirs and net connections etc. like we do in the rm soak test ).

Reply:
@grahamwhaley - That sounds good to me!

Reply:
I'd prefer to keep the current implementation and just validate (check no empty values, paths, etc) the configuration toml in `loadKataConfiguration` method

Reply:
^ @grahamwhaley and @jodh-intel

Reply:
@devimc can you expand on that. IIUC the intent of this PR, Ideally we should be checking all system level sanity items we can, including things that are not in the config file (like the number pods and vms down in /var match our container count, and that we have the correct number of file system and network mounts/connections etc.).
The majority of that checking already exists over at [the rm soak test](https://github.com/kata-containers/tests/blob/master/integration/stability/soak_parallel_rm.sh#L55-L130). My idea is that we do that rigorous system sanity check after every test, to catch dangling issues early.